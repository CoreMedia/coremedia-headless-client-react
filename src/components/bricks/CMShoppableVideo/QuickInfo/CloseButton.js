// @flow
import React from 'react';
import PropTypes from 'prop-types';

import { IconButton as BasicIconButton } from '../../../basic/Button';
import CloseIcon from '../../../basic/Icon/CloseIcon';

const IconButton = BasicIconButton.extend`
  position: absolute;
  top: 5px;
  right: 5px;
`;

IconButton.displayName = 'IconButton';

type Props = {
  handleClick: (ev: SyntheticEvent<HTMLButtonElement>) => void,
};

const CloseButton = ({ handleClick }: Props) => (
  <IconButton onClick={handleClick} color="#808080">
    <CloseIcon />
  </IconButton>
);

CloseButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default CloseButton;
