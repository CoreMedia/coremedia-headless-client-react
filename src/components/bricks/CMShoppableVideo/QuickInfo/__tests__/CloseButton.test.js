// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import CloseButton from '../CloseButton';

describe('CloseButton Container', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<CloseButton handleClick={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should execute onClick handler', async () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<CloseButton handleClick={handleClick} />);
    wrapper.simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
});
