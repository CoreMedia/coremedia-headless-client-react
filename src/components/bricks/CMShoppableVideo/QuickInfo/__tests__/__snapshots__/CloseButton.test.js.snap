// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton Container should render correctly 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  padding: 7px;
  margin: 0;
  border-style: none;
  border-radius: 50%;
  border-width: 0;
  outline: none;
  color: #808080;
  background-color: transparent;
  cursor: pointer;
  position: absolute;
  top: 5px;
  right: 5px;
}

<button
  className="c0"
  color="#808080"
  onClick={[Function]}
>
  <svg
    aria-labelledby="ID"
    className="c1"
    role="img"
    size={undefined}
    viewBox="0 0 512 512"
  >
    <title
      id="ID"
    >
      close
    </title>
    <path
      d="M256 0C114.624 0 0 114.624 0 256s114.624 256 256 256 256-114.624 256-256S397.376 0 256 0zm126.624 337.376l-45.248 45.248L256 301.248l-81.376 81.376-45.248-45.248L210.752 256l-81.376-81.376 45.248-45.248L256 210.752l81.376-81.376 45.248 45.248L301.248 256l81.376 81.376z"
      fill="currentColor"
    />
  </svg>
</button>
`;
